index.bs

```diff
@@ -247,6 +247,199 @@ a generator to a processor (e.g., with a call like
 the Web Platform-exposed signals, but ignores any internal custom signals as there is no
 way for the platform sinks to know what the upstream source is.
 
+## MediaStreamTrack ## {#generator-as-track}
+A {{MediaStreamTrackGenerator}} is a {{MediaStreamTrack}}. This section adds
+clarifications on how a {{MediaStreamTrackGenerator}} behaves as a
+{{MediaStreamTrack}}.
+
+### clone ### {#generator-clone}
+The {{MediaStreamTrack/clone}} method on a {{MediaStreamTrackGenerator}}
+returns a new {{MediaStreamTrack}} object whose underlying source is the
+same as the one for the {{MediaStreamTrackGenerator}} being cloned.
+This underlying source is the {{MediaStreamTrackGenerator/writable}} of
+the {{MediaStreamTrackGenerator}}.
+
+### stop ### {#generator-stop}
+The {{MediaStreamTrack/stop}} method on a {{MediaStreamTrackGenerator}} stops
+the track. When the last track connected to
+the {{MediaStreamTrackGenerator/writable}} of a {{MediaStreamTrackGenerator}} is
+stopped, the {{MediaStreamTrackGenerator/writable}} is closed.
+
+### Constraintable properties ### {#generator-constrainable-properties}
+
+The following constrainable properties are defined for video
+{{MediaStreamTrackGenerator}}s and any {{MediaStreamTrack}}s sourced from
+a {{MediaStreamTrackGenerator}}:
+<table>
+  <thead>
+    <tr>
+      <th>
+        Property Name
+      </th>
+      <th>
+        Values
+      </th>
+      <th>
+        Notes
+      </th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr id="def-constraint-width">
+      <td data-tests="">
+        <dfn>width</dfn>
+      </td>
+      <td>
+        {{ConstrainULong}}
+      </td>
+      <td>
+        As a setting, this is the width, in pixels, of the latest
+        frame received. As a capability, max MUST reflect the largest width
+        with which a VideoFrame object can be created, and min MUST reflect
+        the smallest width with which a VideoFrame can be created.
+      </td>
+    </tr>
+    <tr id="def-constraint-height">
+      <td data-tests="">
+        <dfn>height</dfn>
+      </td>
+      <td>
+        {{ConstrainULong}}
+      </td>
+      <td>
+        As a setting, this is the height, in pixels, of the latest
+        frame received. As a capability, `max` MUST reflect the largest height
+        with which a {{VideoFrame}} object can be created, and `min` MUST reflect
+        the smallest height with which a VideoFrame can be created.
+      </td>
+    </tr>
+    <tr id="def-constraint-frameRate">
+      <td data-tests="">
+        <dfn>frameRate</dfn>
+      </td>
+      <td>
+        {{ConstrainDouble}}
+      </td>
+      <td>
+        As a setting, this is an estimate of the frame rate based on
+        recently received frames. As a capability `min` MUST be zero and
```

Comment:
Is it based on the timestamps of recently received frames or the actual rate of frames being written to writable?

Reply:
Update the text to reflect that this is based on frames delivered by the track to its sinks, since they may have been resized by the track.

Reply:
Sorry, I think my question was unclear. A VideoFrame or AudioFrame can be created with an arbitrary timestamp, which may be unrelated to the actual time. The duration between when two frames are sent to the track could be quite a bit different from the differences between their timestamps. Will the framerate for the track be based on the difference between frame timestamps or based on the frequency with which frames are written to the writable stream?

Reply:
I wrote the text saying it is the frequency with which they're written  to the track sinks. In many cases this is the same as the frequency with which they're written to the writable stream, but for video it can be different due to some frames being dropped to satisfy the frameRate constraint.